### ISSUE TITLE
Issue found in/using golangci-lint, about warnings not processing properly

### ISSUE THREAD

**Issue by mrz1836** [kkHAIKE/contextcheck#21]
Please see this issue:

https://github.com/golangci/golangci-lint/issues/4550

We have concluded it's an issue in `contextcheck` but I do not know specifically what is going on. There is example code and changes made that can reproduce the issue:

https://github.com/golangci/golangci-lint/issues/4550#issuecomment-2009738678

**Comment by kkHAIKE** [comment]
it's fix: https://github.com/golangci/golangci-lint/pull/4552

**Comment by ldez** [comment]
@kkHAIKE It's a `contextcheck` bug, I just provided a workaround.

`contextcheck` should not report issues without a filepath.


```console
$ ./contextcheck ./...
-: Function `NewWriteLock$1` should pass the context parameter
```
